pytest.ini

```diff
@@ -4,4 +4,23 @@ addopts = --failed-first
 testpaths =
     tests
 filterwarnings =
-    ignore::UserWarning
+    # iris
+    ignore:Ignoring config item 'Resources':UserWarning
+    ignore:`np.bool` is a deprecated alias for the builtin `bool`:DeprecationWarning
+    ignore:`np.float` is a deprecated alias for the builtin `float`:DeprecationWarning
+    # xarray
+    ignore:dropping labels using list-like labels is deprecated:DeprecationWarning
+    # fsspec
+    ignore:SelectableGroups dict interface is deprecated:DeprecationWarning
+    # pyke
+    ignore:the imp module is deprecated in favour of importlib:DeprecationWarning
+    # geonum
+    ignore:Plotting of maps etc. is deactivated:UserWarning
+    # scipy.stats PearsonRConstantInputWarning and SpearmanRConstantInputWarning
+    ignore:An input array is constant:RuntimeWarning
+    # np.divide
+    ignore:invalid value encountered in true_divide:RuntimeWarning
+    # mathutils.weighted_corr
+    ignore:invalid value encountered in double_scalars:RuntimeWarning
+    # helpers.resample_timeseries
+    ignore:Mean of empty slice:RuntimeWarning
```

Comment:
Don't we want to see some of the warnings? Like DeprecationWarning? So we can fix it before it breaks?

Reply:
Fair point, but those `DeprecationWarning`s are triggered on `iris` not `pyaerocom`.

I resolved several warning generated on `pyaerocom`.
These are warning that are generated on imported dependencies,
and some that can not be resolved  without some refactoring.

Maybe, I could wrap the pyaerocom offending lines on a `with warn.ignore` clause.
Not sure what to do about the warnings triggered on imported modules.

Reply:
I encapsulated all the different warnings on the calls from the test_functions and the module imports.
Solved some `DeprecationWarning`s I have not seen before and set all warnings to errors, so any new warning will trigger the tests to fail.

Reply:
there are some different warnings on Python 3.8 and 3.9 that I can not solve right now, so I'll allow them back